src/csim/update_ops_control_single_target_multi.cpp

```diff
@@ -63,3 +68,81 @@ void single_qubit_control_multi_qubit_dense_matrix_gate(
     free(buffer);
     free(matrix_mask_list);
 }
+
+#ifdef _USE_MPI
+void single_qubit_control_multi_qubit_dense_matrix_gate_mpi(
+    UINT control_qubit_index, UINT control_value,
+    const UINT* target_qubit_index_list, UINT target_qubit_index_count,
+    const CTYPE* matrix, CTYPE* state, ITYPE dim, UINT inner_qc) {
+    UINT* new_target_index_list =
+        (UINT*)malloc((size_t)(sizeof(UINT) * target_qubit_index_count));
+
+    // count outer-target qubit
+    UINT num_outer_target = 0;
+    for (UINT i = 0; i < target_qubit_index_count; ++i) {
+        new_target_index_list[i] = target_qubit_index_list[i];
+        if (target_qubit_index_list[i] >= inner_qc) ++num_outer_target;
+    }
+
```

Comment:
Throw exception if num_outer_target > inner_qc.

Reply:
Thank you for pointing it out. I add an exception like the one above.